scenes/game/local_game.gd

```diff
@@ -1412,6 +1430,10 @@ class Player:
 		if parent.active_strike:
 			# Undo continuous effects
 			for effect in _find_during_strike_effects(card):
+				if not buddy_ignore_condition and not parent.is_effect_condition_met(self, effect, null):
+					# Only undo effects that were given in the first place.
+					continue
```

Comment:
the shovel knight unit tests pass with this right? i really hope there aren't any mid-strike-removable cards whose conditions can also change mid-strike

Reply:
Yep, all shield/shovel pass.